sessionctx/binloginfo/binloginfo.go

```diff
@@ -131,11 +132,27 @@ func (info *BinlogInfo) WriteBinlog(clusterID uint64) error {
 }
 
 // SetDDLBinlog sets DDL binlog in the kv.Transaction.
-func SetDDLBinlog(client *pumpcli.PumpsClient, txn kv.Transaction, jobID int64, ddlQuery string) {
+func SetDDLBinlog(client *pumpcli.PumpsClient, txn kv.Transaction, job *model.Job, ddlQuery string) {
 	if client == nil {
 		return
 	}
 
+	//append default charset and collate
```

Comment:
How about extract below logic to a single function and add a unit test for the function.

Reply:
Roger that!